variables.tf

```diff
@@ -1,24 +1,31 @@
 variable "hub_virtual_networks" {
   description = "A map of hub virtual networks to create"
   type = map(object({
+    ## As this variable is a map[string]VirtualNetwork, do we need vnet's name here? Would using the key as vnet's name be better?
```

Comment:
Due constraint that map key is known at plan, there maybe a scenario where a caller uses other resource data to generate the vnet names, there I propose keeping vnet name distinct from map key.